### ISSUE TITLE
Add an option to set a custom folder for the --git-diff- commands

### ISSUE THREAD

**Issue by Glutamat42** [infection/infection#2007]
**Is your feature request related to a problem? Please describe.**
I am using infection on a moodle plugin and now want to use the git-diff-lines/filter feature. The issue is that the project root (where i have to execute the phpunit tests / infection from) and the git working directory are not the same.

The folder structure looks as follows:
- moodle
  - folder_a
  - folder_b
  - local
    - `<my plugin>`
   
I have to run the test with `moodle` being the working directory and then instruct phpunit to either run a subset of the tests or use the phpunit.xml file in the `moodle/local/<my plugin>` folder. `moodle/local/<my pluign>` is also the git repo. The issue is that i cannot tell infection that it has to use `moodle/local/<my plugin>` as the working directory for git.

**Describe the solution you'd like**
A command line or infection.json5 switch "git-working-directory"

**Describe alternatives you've considered**

**Additional context**

**Comment by maks-rafalko** [comment]
Hey, could you please try to install Infection locally as a composer dependency (or as a global package) to be able to update the source code and try to pass your folder to `-C` option inside `git diff`? 

Just to check if it will work and if this is the only one issue we have to improve to achieve what you need.

So, at the low level you need `git -C local/<my plugin> diff ...`

You can do it by updating this line of code:

https://github.com/infection/infection/blob/e30db7dda2e9308f1435097b56d20e959b06db17/src/Logger/GitHub/GitDiffFileProvider.php#L69-L75

so basically you will need to do something like this:

```diff
'git',
+ '-C',
+ 'local/<my plugin>',
'diff',
$referenceCommit,
'--diff-filter',
$gitDiffFilter,
'--name-only',
'--',
```

**Comment by Glutamat42** [comment]
The suggested change worked perfectly for --git-diff-filter. For filter-lines, it required an additional tweak. It identified only a few files were modified, but failed with a FilesystemException (see log below). Adding `local/adlersetup/` as a prefix to all lines resolved the issue.

```
    public function provideWithLines(string $gitDiffBase): string
    {
        $referenceCommit = $this->findReferenceCommit($gitDiffBase);

        $filter = $this->shellCommandLineExecutor->execute([
            'git',
            '-C',
            'local/adlersetup',
            'diff',
            $referenceCommit,
            '--unified=0',
            '--diff-filter=AM',
        ]);
        $lines = explode(PHP_EOL, $filter);
        $lines = array_filter($lines, static fn ($line): bool => preg_match('/^(\\+|-|index)/', $line) === 0);
        $lines = array_map(static fn ($line): string => 'local/adlersetup/' . $line, $lines);

        return implode(PHP_EOL, $lines);
    }
```


```
/..../infection/build/infection.phar -s --only-covered --configuration=local/adlersetup/infection.json5 --git-diff-lines --git-diff-base=origin/main

Running initial test suite...

PHPUnit version: 9.6.21

  148 [============================] 52 secs

Generate mutants...

Processing source code files: 0/2
In FilesystemException.php line 11:
                    
  An error occured  
```

**Comment by theofidry** [comment]
I'm a bit confused by the setup.

Do I understand correctly that the problem is that you would like to use `--git-diff-lines` but further restrict the sources to a sub-directory?

As a workaround this should already be possible by limiting `source` in the configuration, as anything outside of it is removed.
